packages/veraswap-app/src/atoms/uniswap.ts

```diff
@@ -4,54 +4,99 @@ import {
+import { numberToHex, zeroAddress } from "viem";
 import { UNISWAP_CONTRACTS } from "@owlprotocol/veraswap-sdk/constants";
 import { queryOptions } from "@tanstack/react-query";
 import { currencyInAtom, currencyOutAtom, tokenInAmountAtom } from "./tokens.js";
 import { disabledQueryOptions } from "./disabledQuery.js";
-import { config } from "@/config.js";
+import { config, chains } from "@/config.js";
 
 // const emptyToken = new Token(1, zeroAddress, 1);
 // const emptyCurrencyAmount = CurrencyAmount.fromRawAmount(emptyToken, 1);
 
 //TODO: As tanstack query so that it refreshes
+// export const routeMultichainAtom = atomWithQuery((get) => {
+//     const queryClient = get(queryClientAtom);
+//     const currencyIn = get(currencyInAtom);
+//     const currencyOut = get(currencyOutAtom);
+//     const exactAmount = get(tokenInAmountAtom);
+
+//     if (!currencyIn || !currencyOut || !exactAmount) return disabledQueryOptions as any;
+
+//     const currencyInAddress = getUniswapV4Address(currencyIn);
+//     const currencyOutAddress = getUniswapV4Address(currencyOut);
+
+//     return queryOptions({
+//         queryFn: async () => {
+//             const route = await getRouteMultichain(queryClient, config, {
+//                 currencyIn,
+//                 currencyOut,
+//                 exactAmount,
+//                 contractsByChain: UNISWAP_CONTRACTS,
+//                 currencyHopsByChain: {},
+//             });
+
+//             console.debug({ route });
+
+//             return route;
+//         },
+//         queryKey: [
+//             "getRouteMultichain",
+//             currencyIn.chainId,
+//             currencyInAddress,
+//             currencyOut.chainId,
+//             currencyOutAddress,
+//             numberToHex(exactAmount),
+//         ],
+//     });
+// }) as Atom<AtomWithQueryResult<Awaited<ReturnType<typeof getRouteMultichain>>>>;
+
 export const routeMultichainAtom = atomWithQuery((get) => {
     const queryClient = get(queryClientAtom);
     const currencyIn = get(currencyInAtom);
     const currencyOut = get(currencyOutAtom);
-    const exactAmount = get(tokenInAmountAtom);
+    const amountIn = get(tokenInAmountAtom);
 
-    if (!currencyIn || !currencyOut || !exactAmount) return disabledQueryOptions as any;
+    if (!currencyIn || !currencyOut || !amountIn) return disabledQueryOptions as any;
 
     const currencyInAddress = getUniswapV4Address(currencyIn);
     const currencyOutAddress = getUniswapV4Address(currencyOut);
 
     return queryOptions({
         queryFn: async () => {
-            const route = await getRouteMultichain(queryClient, config, {
+            const contractsByChain = Object.fromEntries(
+                chains.map((chain) => [
+                    chain.id,
+                    {
+                        weth9: UNISWAP_CONTRACTS[chain.id]?.weth9 ?? zeroAddress,
+                        metaQuoter: UNISWAP_CONTRACTS[chain.id]?.metaQuoter ?? zeroAddress,
```

Comment:
This should be done in getRouteMultichain. Also, if metaQuoter is not defined, our quoter should just ignore that chain. So it should be an optional param. Haven't checked weth9 usage, but we should also ignore weth if it's not defined